### PR TITLE
Filter

### PR DESCRIPTION
**Pull request by heartjigi**
Add Filter feature

### CODE REVIEW

File: src/components/MyPage.js [hunk 1]
```diff
@@ -110,7 +110,8 @@ class MyPage extends Component {
   signOut() {
     try {
       UserUtil.signOut((result, error) => {
-        AsyncStorage.removeItem('token', () => { Actions.login(); });
+        let keys = ['token', 'filter'];
```
Comment (SarahhhC) [inline src/components/MyPage.js:113]: Reset filter when sign out? Why?
Didn't back-end side save user filter information and adjust it?

File: src/components/SignUp/CareerInfo.js [hunk 2]
```diff
@@ -115,9 +117,16 @@ class CareerInfo extends Component {
       }
     }
 
+    for (let i = 0; i < this.state.pressed.length; i++)
```
Comment (SarahhhC) [inline src/components/SignUp/CareerInfo.js]: add { }

File: src/components/SignUp/CareerInfo.js [hunk 3]
```diff
@@ -115,9 +117,16 @@ class CareerInfo extends Component {
...
+      this.state.pressed[i] = false;
     this.forceUpdate();
   }
 
+  onPress() {
+    for (let i = 0; i < this.state.pressed.length; i++)
+      this.state.pressed[i] = true;
```
Comment (SarahhhC) [inline src/components/SignUp/CareerInfo.js:129]: Can you explain what this is? `state.pressed` for

File: src/components/SignUp/CareerInfo.js [hunk 4]
```diff
@@ -153,21 +162,23 @@ class CareerInfo extends Component {
...
-                    clear={clear}
-                    activate={activate}
-                    index={idx}
-                    width={Dimensions.get('window').width - 60}
-                    defaultValue={' '}
-                    optionListRef={this.getOptionList.bind(this)}
-                    onSelect={this.onSelect.bind(this)}>
-                    {this.getOptionSet(idx)}
-                  </Select>
+                <View key={idx} style={[styles.questionContainer, { zIndex: zIdx }]}>
```
Comment (SarahhhC) [inline src/components/SignUp/CareerInfo.js]: check indentation

File: src/components/Main.js [hunk 5]
```diff
@@ -181,43 +182,45 @@ class Main extends Component {
...
-        <Activity
-          tabLabel="ios-people"
-          style={styles.tabView}
-          currentActivityPage={this.state.currentActivityPage}
-          me={this.props.me}
-        />
-        <ChannelList tabLabel="ios-chatbubbles" style={styles.tabView} me={this.props.me} />
-        <MyPage tabLabel="md-contact" me={this.props.me} />
-      </ScrollableTabView>
+          <ScrollableTabView
```
Comment (SarahhhC) [inline src/components/Main.js]: check indentation.

File: src/components/SignUp/SignUpMETA.js [hunk 6]
```diff
@@ -128,21 +128,32 @@ export const CareerData = {
...
+export const OptionsFilter = [
+    'Study abroad',
+    'Career advice',
     'Portfolio & Resume',
-    'Start up',
+    'Startup',
     'Career change',
     'Networking',
-    'Soft skills (e.g., Communication..)',
+    'Soft skills',
```
Comment (SarahhhC) [inline src/components/SignUp/SignUpMETA.js:149]: Why options and option filter's contents are different? No relation with them?

File: src/components/SignUp/SignUpMETA.js [hunk 7]
```diff
@@ -128,21 +128,32 @@ export const CareerData = {
...
 ];
 
 export const Personalites = [
-    ['Extroverts', 'Introverts'],
+    ['Extrovert', 'Introvert'],
     ['Rational', 'Intuitive'],
-    ['Feelers', 'Thinkers'],
-    ['Judgers', 'Perceivers'],
+    ['Feeler', 'Thinker'],
+    ['Judger', 'Perceiver'],
```
Comment (SarahhhC) [inline src/components/SignUp/SignUpMETA.js:156]: Personalities changed? Do database need update? I think yes...
Please tell me about these things 😄

File: src/components/UserList/UserList.js [hunk 8]
```diff
@@ -37,13 +39,13 @@ class UserList extends Component {
     if (error) {
       alert(error);
     } else if (result) {
+
```
Comment (SarahhhC) [inline src/components/UserList/UserList.js:56]: delete line and where is else part?

File: src/utils/ApiUtil.js [hunk 9]
```diff
@@ -87,12 +87,13 @@ class ApiUtil extends Component {
...
     .catch((error) => {
-      if (JSON.stringify(error) !== '{}') callback(null, JSON.stringify(error));
-      else callback(null, error);
+      if (JSON.stringify(error) === '{}') callback(null, error);
+      else callback(null, JSON.stringify(error));
     });
   }
 
   getResponse(response) {
+    console.log(response);
```
Comment (SarahhhC) [inline src/utils/ApiUtil.js]: delete console.

File: src/utils/ApiUtil.js [hunk 10]
```diff
@@ -87,12 +87,13 @@ class ApiUtil extends Component {
     .then(this.getResponse.bind(this))
     .then((res) => callback(res, null))
     .catch((error) => {
-      if (JSON.stringify(error) !== '{}') callback(null, JSON.stringify(error));
-      else callback(null, error);
+      if (JSON.stringify(error) === '{}') callback(null, error);
```
Comment (SarahhhC) [inline src/utils/ApiUtil.js]: let's add braces. { }

File: src/components/MyPage.js (hunk 1 above)
Comment (heartjigi) [reply src/components/MyPage.js:113]: yes, actually we didn't save user's filter data.
So I saved the filter data on local. Let's talk about this issue @KyuaKwon @Stacyyya

File: src/components/SignUp/CareerInfo.js (hunk 3 above)
Comment (heartjigi) [reply src/components/SignUp/CareerInfo.js:129]: The purpose of this state val is activate just one dropdown.

File: src/components/SignUp/CareerInfo.js (hunk 2 above)
Comment (heartjigi) [reply src/components/SignUp/CareerInfo.js]: solved

File: src/components/SignUp/CareerInfo.js (hunk 4 above)
Comment (heartjigi) [reply src/components/SignUp/CareerInfo.js]: solved

File: src/components/SignUp/SignUpMETA.js (hunk 6 above)
Comment (heartjigi) [reply src/components/SignUp/SignUpMETA.js:149]: In filter, we decide delete after e.g. , however, I send data with original string.

File: src/components/SignUp/SignUpMETA.js (hunk 7 above)
Comment (heartjigi) [reply src/components/SignUp/SignUpMETA.js:156]: Yeah, include this. we have to update datas, It was changed with Stacy

File: src/components/UserList/UserList.js (hunk 8 above)
Comment (heartjigi) [reply src/components/UserList/UserList.js:56]: this line for comments line (JSCS),
and this part is not related me...

File: src/utils/ApiUtil.js (hunk 9 above)
Comment (heartjigi) [reply src/utils/ApiUtil.js]: solved

File: src/components/Main.js (hunk 5 above)
Comment (heartjigi) [reply src/components/Main.js]: solved

File: src/components/SignUp/SignUpMETA.js (hunk 6 above)
Comment (SarahhhC) [reply src/components/SignUp/SignUpMETA.js:149]: ok good!

File: src/components/SignUp/SignUpMETA.js (hunk 7 above)
Comment (SarahhhC) [reply src/components/SignUp/SignUpMETA.js:156]: Ok, let me know about them!

File: src/components/UserList/UserList.js (hunk 8 above)
Comment (SarahhhC) [reply src/components/UserList/UserList.js:56]: Ok!

File: ios/Bridgeme.xcodeproj/project.pbxproj [hunk 11]
```diff
@@ -52,6 +52,7 @@
 		D260393C1DF7D9D0004DC6C9 /* SF-UI-Text-Light.otf in Resources */ = {isa = PBXBuildFile; fileRef = D26039371DF7D8E0004DC6C9 /* SF-UI-Text-Light.otf */; };
```
Comment (ace0625) [inline ios/Bridgeme.xcodeproj/project.pbxproj:52]: No need to upload this file in this PR

File: src/components/UserList/UserList.js [hunk 12]
```diff
@@ -58,8 +60,8 @@ class UserList extends Component {
     }
   }
 
-  componentDidMount() {
-    UserUtil.getMentorList(this.onServerCallback.bind(this));
+  componentWillReceiveProps(props) {
```
Comment (ace0625) [inline src/components/UserList/UserList.js:87]: why did you delete componentDidMount? I believe it should be there.

File: src/utils/CheckBox.js [hunk 13]
```diff
@@ -20,15 +20,20 @@ class CheckBox extends Component {
 
   getIcon() {
     if (this.props.checked)
```
Comment (ace0625) [inline src/utils/CheckBox.js]: use curly brackets for if statements

File: src/utils/CheckBox.js [hunk 14]
```diff
@@ -20,15 +20,20 @@ class CheckBox extends Component {
...
+        size={(this.props.iconSize) ? this.props.iconSize : 20} />);
     else
       return;
   }
 
   render() {
     let backColor = this.props.checked  ? '#fafafa' : '#ffffff';
-
-    const { iconStyle, labelStyle } = this.props;
+    const {
```
Comment (ace0625) [inline src/utils/CheckBox.js:32]: It would be better set this const part out of render function.

File: src/utils/CheckBox.js (hunk 13 above)
Comment (heartjigi) [reply src/utils/CheckBox.js]: solved

File: src/utils/CheckBox.js (hunk 14 above)
Comment (heartjigi) [reply src/utils/CheckBox.js:32]: actually this style is from react native dropdown module.
if I change this style, there are many jobs.
Do you think this work is needed?

**Comment by ifndefdeadmau5** [comment]
LGTM and I'd like to suggest you write commit message more meaningfull and also it would be good if there is a single change at one commit. or at least you can explain about your pull request at the content. Thanks!

[Approved with PullApprove](https://pullapprove.com/sv-bootcamp/bridgeme-client/pull-request/133/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=sv-bootcamp/bridgeme-client)

**Comment by heartjigi** [comment]
I try to write commit message more meaningfull!
Thanks!
